### ISSUE TITLE
1.7 death crash - ArrayIndexOutOfBoundsException: 4

### ISSUE THREAD

**Issue by franxyk** [ViaVersion/ViaRewind#306]
**Describe the bug, provide any errors**
On use ViaRewind on sv 1.16, enter mc client 1.7, when dying the client crashes

**How can we reproduce it?**
1. Start spigot server 1.16.x
2. Login on 1.7 client'
3. Death on any reason 

**Expected behaviour**
Spawn on death screen, without crashing the mc client.

**Screenshots**
none

**ViaVersion Dump:**
https://dump.viaversion.com/006b92741d70375f0d0d5856d956692e6cb5720a2e8306fcac74c20a6803e452

**Additional server info**
Only use viaversion, Viabackwards and ViaRewind 

**Checklist** (mark with [X] to check)
- [x] I have included a ViaVersion dump
- [] If applicable, I have included a paste (not a screenshot) of the error
- [] I have tried a build from https://ci.viaversion.com/ and the issue still persists
- [] I have looked at https://www.spigotmc.org/resources/viarewind-legacy-support.52924/ and it does not offer a fix for my problem

**IMPORTANT: Follow the Checklist or we can't reproduce your issue and your issue will be close**

**Comment by Camotoy** [comment]
Please fill out the bug report given to it and don't delete its contents. Thank you!

**Comment by franxyk** [comment]
> 
> 
> Please fill out the bug report given to you and don't delete its contents. Thank you!

clever

**Comment by Jo0001** [comment]
Crash message: `java.lang.ArrayIndexOutOfBoundsException: 4`
(Latest Paper 1.16.5 + Via ci builds)

**Comment by HEROOSTECH** [comment]
Use ur fuck*ng eyes and fix these issues. ViaBackwards and ViaVersion have good support, but support of this fuck*d ViaRewind is very bad. Fix these issues or archive project.

**Comment by creeper123123321** [comment]
i can reproduce by joining geyser 1.16 test server with 1.7.10 client

```
---- Minecraft Crash Report ----
// Don't be sad, have a hug! <3

Time: 8/31/22 3:01 PM
Description: Unexpected error

java.lang.ArrayIndexOutOfBoundsException: 4
	at yz.c(SourceFile:1633)
	at bjb.a(SourceFile:827)
	at hz.a(SourceFile:39)
	at hz.a(SourceFile:9)
	at ej.a(SourceFile:174)
	at bje.e(SourceFile:212)
	at bao.p(SourceFile:1306)
	at bao.ak(SourceFile:774)
	at bao.f(SourceFile:728)
	at net.minecraft.client.main.Main.main(SourceFile:148)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.polymc.impl.OneSixLauncher.invokeMain(OneSixLauncher.java:104)
	at org.polymc.impl.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:175)
	at org.polymc.impl.OneSixLauncher.launch(OneSixLauncher.java:185)
	at org.polymc.EntryPoint.listen(EntryPoint.java:144)
	at org.polymc.EntryPoint.main(EntryPoint.java:74)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
	at yz.c(SourceFile:1633)
	at bjb.a(SourceFile:827)
	at hz.a(SourceFile:39)
	at hz.a(SourceFile:9)
	at ej.a(SourceFile:174)
	at bje.e(SourceFile:212)

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [bjk['creeper123123321'/35103888, l='MpServer', x=156.95, y=55.62, z=-143.17]]
	Chunk stats: MultiplayerChunkCache: 529, 529
	Level seed: 0
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: 
	Level spawn location: World: (188,63,-124), Chunk: (at 12,3,4 in 11,-8; contains blocks 176,0,-128 to 191,255,-113), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,0,-512 to 511,255,-1)
	Level time: 162917 game time, 1328223181 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
	Forced entities: 47 total; [ws['Cod'/35104670, l='MpServer', x=180.69, y=11.00, z=-119.19], yq['Zombi'/35104794, l='MpServer', x=187.50, y=19.00, z=-133.50], yq['Zombi'/35104792, l='MpServer', x=193.06, y=60.00, z=-160.50], bjk['creeper123123321'/35103888, l='MpServer', x=156.95, y=55.62, z=-143.17], yq['Zombi'/35104821, l='MpServer', x=148.50, y=11.00, z=-143.50], yq['Zombi'/35104820, l='MpServer', x=141.28, y=19.00, z=-132.70], we['Parrot'/35104688, l='MpServer', x=124.50, y=69.00, z=-177.38], ws['Tropical Fish'/35104703, l='MpServer', x=125.55, y=58.90, z=-132.36], ws['Tropical Fish'/35104702, l='MpServer', x=127.00, y=58.58, z=-132.94], wn['Turtle'/1567806, l='MpServer', x=153.03, y=63.00, z=-99.91], ws['Tropical Fish'/35104701, l='MpServer', x=127.41, y=59.28, z=-130.47], ws['Cod'/35104826, l='MpServer', x=173.09, y=53.00, z=-118.06], ws['Tropical Fish'/35104824, l='MpServer', x=141.34, y=59.88, z=-107.06], ws['Calamar'/35104679, l='MpServer', x=136.19, y=49.72, z=-99.97], we['Parrot'/7575233, l='MpServer', x=201.09, y=64.00, z=-154.22], wi['Caballo Esqueleto'/35104676, l='MpServer', x=168.91, y=64.00, z=-169.91], wi['Caballo Esqueleto'/35104675, l='MpServer', x=163.47, y=64.00, z=-154.00], wi['Caballo Esqueleto'/35104674, l='MpServer', x=158.34, y=61.00, z=-161.96], xi['Bote'/35104673, l='MpServer', x=175.84, y=62.81, z=-170.94], ws['Cod'/35104801, l='MpServer', x=165.67, y=58.07, z=-143.44], yq['Zombi'/35104672, l='MpServer', x=161.50, y=18.00, z=-115.50], ws['Calamar'/35104685, l='MpServer', x=124.78, y=50.03, z=-101.41], ws['Tropical Fish'/35104812, l='MpServer', x=123.56, y=62.00, z=-112.63], ws['Cod'/35104809, l='MpServer', x=118.50, y=62.75, z=-136.47], ws['Cod'/35104808, l='MpServer', x=137.66, y=59.91, z=-106.34], yl['Esqueleto'/35104727, l='MpServer', x=151.22, y=25.00, z=-156.50], yl['Esqueleto'/35104854, l='MpServer', x=126.50, y=23.00, z=-96.50], we['Murciélago'/35104731, l='MpServer', x=127.25, y=23.10, z=-116.69], wn['Turtle'/1567942, l='MpServer', x=157.88, y=60.00, z=-117.22], ws['Cod'/35104709, l='MpServer', x=136.51, y=49.28, z=-150.21], ws['Cod'/35104708, l='MpServer', x=135.89, y=49.22, z=-149.31], ws['Cod'/35104707, l='MpServer', x=138.46, y=53.03, z=-149.45], zc['Flecha'/35104835, l='MpServer', x=154.19, y=48.03, z=-140.75], ws['Tropical Fish'/35104704, l='MpServer', x=126.60, y=58.65, z=-132.42], we['Murciélago'/35104719, l='MpServer', x=128.44, y=23.10, z=-111.44], ws['Tropical Fish'/35104847, l='MpServer', x=109.50, y=43.78, z=-99.59], ws['Cod'/35104718, l='MpServer', x=138.84, y=61.24, z=-106.47], xz['Creeper'/35104846, l='MpServer', x=174.50, y=63.00, z=-126.50], yq['Drowned'/1615682, l='MpServer', x=183.50, y=45.71, z=-123.18], zc['Flecha'/35104845, l='MpServer', x=152.75, y=46.03, z=-136.75], wn['Turtle'/1615683, l='MpServer', x=177.53, y=62.00, z=-112.28], yq['Zombi'/35104844, l='MpServer', x=188.47, y=62.00, z=-108.69], yl['Esqueleto'/35104759, l='MpServer', x=156.30, y=54.90, z=-142.68], yl['Esqueleto'/35104756, l='MpServer', x=156.50, y=31.00, z=-146.50], yq['Zombi'/35104764, l='MpServer', x=204.50, y=25.00, z=-136.50], zc['Flecha'/35104870, l='MpServer', x=156.26, y=56.72, z=-142.64], we['Murciélago'/35104865, l='MpServer', x=115.24, y=19.06, z=-123.15]]
	Retry entities: 0 total; []
	Server brand: Paper (VIAaaS C: 1.7.6-1.7.10 (5) S: 1.16.4/5 (754))
	Server type: Non-integrated multiplayer server
Stacktrace:
	at bjf.a(SourceFile:289)
	at bao.b(SourceFile:1972)
	at bao.f(SourceFile:742)
	at net.minecraft.client.main.Main.main(SourceFile:148)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.polymc.impl.OneSixLauncher.invokeMain(OneSixLauncher.java:104)
	at org.polymc.impl.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:175)
	at org.polymc.impl.OneSixLauncher.launch(OneSixLauncher.java:185)
	at org.polymc.EntryPoint.listen(EntryPoint.java:144)
	at org.polymc.EntryPoint.main(EntryPoint.java:74)

-- System Details --
Details:
	Minecraft Version: 1.7.10
	Operating System: Linux (amd64) version 5.15.0-46-generic
	Java Version: 1.8.0_332, Oracle Corporation
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 163470520 bytes (155 MB) / 443023360 bytes (422 MB) up to 954728448 bytes (910 MB)
	JVM Flags: 2 total; -Xms128m -Xmx1024m
	AABB Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
	IntCache: cache: 0, tcache: 0, allocated: 13, tallocated: 95
	Launched Version: 1.7.10
	LWJGL: 2.9.4
	OpenGL: Mesa Intel(R) UHD Graphics 620 (WHL GT2) GL version 4.6 (Compatibility Profile) Mesa 21.3.9 (git-78c96ae5b6), Intel
	GL Caps: Using GL 1.3 multitexturing.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Anisotropic filtering is supported and maximum anisotropy is 16.
Shaders are available because OpenGL 2.1 is supported.

	Is Modded: Probably not. Jar signature remains and client brand is untouched.
	Type: Client (map_client.txt)
	Resource Packs: []
	Current Language: Español (Argentina)
	Profiler Position: N/A (disabled)
	Vec3 Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
	Anisotropic Filtering: Off (1)
```

**Comment by creeper123123321** [comment]
method_7596 -> ClientPlayPacketListener.onEquipmentUpdate
